common/lib/container-definitions/src/fluidModule.ts

```diff
@@ -3,8 +3,9 @@
  * Licensed under the MIT License.
  */
 
-import { IFluidObject, IProvideFluidCodeDetailsComparer } from "@fluidframework/core-interfaces";
+import { FluidObject, IFluidObject, IProvideFluidCodeDetailsComparer } from "@fluidframework/core-interfaces";
+import { IRuntimeFactory } from "./runtime";
 
 export interface IFluidModule {
-    fluidExport: IFluidObject & Partial<Readonly<IProvideFluidCodeDetailsComparer>>;
+    fluidExport: IFluidObject & FluidObject<IRuntimeFactory & IProvideFluidCodeDetailsComparer>;
```

Comment:
Should this be something like this?

```ts
fluidExport: IFluidObject & Partial<Readonly<IProvideFluidCodeDetailsComparer>> & FluidObject<IRuntimeFactory & IProvideFluidCodeDetailsComparer>;
```

Reply:
FluidObject is basically the same as Partial<Readonly, it just requires the props be self-referential. these packages have type version tests, so this change is fully duck type compatible, otherwise those tests would detect it